xarray/backends/api.py

```diff
@@ -15,34 +15,40 @@
 from ..core.combine import auto_combine
 from ..core.utils import close_on_error, is_remote_uri
 from ..core.pycompat import basestring, path_type
+from ..core.options import OPTIONS
 
 DATAARRAY_NAME = '__xarray_dataarray_name__'
 DATAARRAY_VARIABLE = '__xarray_dataarray_variable__'
 
 
+def _file_startswith(path, *args):
+    """Check that the file statswith with the sepcific byte sequence."""
+    with open(path, 'rb') as f:
+        for b in args:
+            sig = f.read(len(b))
+            print(sig, b)
+            return True if sig == b else False
+
+
 def _get_default_engine(path, allow_remote=False):
     if allow_remote and is_remote_uri(path):  # pragma: no cover
-        try:
-            import netCDF4
-            engine = 'netcdf4'
-        except ImportError:
-            try:
-                import pydap
-                engine = 'pydap'
-            except ImportError:
-                raise ValueError('netCDF4 or pydap is required for accessing '
-                                 'remote datasets via OPeNDAP')
+        for engine in OPTIONS['io_engines']:
+            if engine in ('netcdf4', 'pydap', 'h5netcdf'):
```

Comment:
h5netcdf doesn't work for opendap

Reply:
OPeNDAP is not the only data web service protocol. There could be others in the future, like OGC Web Coverage Service, etc., all with URLs beginning with `http`.

Reply:
Certainly... but as yet h5netcdf/h5py doesn't support any of them :)